### DISCUSSION TITLE
Coverage Resource missing

### DISCUSSION THREAD

**Discussion by cunninghamb505** [microsoft/FHIR-Converter#231] [Q&A: Liquid Engine] [answered]
Hi everyone,
I am trying to ull the IN1 segments and I noticed there is no Coverage Resource. Do I need to create it myself? I am trying with liquid now but confused on how all the pointers and ID's are linked. Any tips or guides out there on how to add segments to the bundle.

**Comment by ranvijaykumar** [comment, marked as answer]
The conversion templates in this repo are based on the 2-to-FHIR [spreadsheet](https://docs.google.com/spreadsheets/d/1PaFYPSSq4oplTvw_4OgOn6h2Bs_CMvCAU9CqC4tPBgk/edit#gid=0). The spreadsheet at present does not provide a mapping between IN1 and Coverage resource. 

So, you will need to make changes in the templates. Familiarity with [Liquid](https://shopify.github.io/liquid/basics/introduction/) templating language will be required. You can either write all your template code in the root template, or in separate template files that you can call from the root template. The default templates are structured in multiple files for the purpose of reuse. If you are using the [VS Code extension](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-health-fhir-converter) for editing the templates, hovering over the sub-template name while pressing CTRL will enable hyperlinks that will take you to the sub-templates.

Here are some tips on how you could add Coverage resource following the same structure as the default templates:

1. Create “Resource/_Coverage.liquid” and “ID/_Coverage.liquid” template files. 
2. Decide on the Coverage Id creation logic. See other templates under the ID folder for examples.
3. See other templates such as _Procedure.liquid for example. 
4. Include "Resource/_Coverage.liquid” and “ID/_Coverage.liquid" from your root template at the appropriate location. See examples of include in ADT_A01.liquid

**Reply by cunninghamb505** [reply]
Thanks Ranvi, 
I posted a video review of what I have so far. Does this look correct? Anything I am missing
https://youtu.be/DbRSwExRi9o

**Reply by cunninghamb505** [reply]
I added some changes and was able to get it to work. Here is an update of it working
https://youtu.be/1dF0gKjoJmI

Still not clear on some of the Evaluate vs include statements , and logic
